integration/integration_test.go

```diff
@@ -9016,3 +9020,216 @@ func testModeratedSessions(t *testing.T, suite *integrationTestSuite) {
+			Kind:    types.KindWindowsDesktop,
+			Version: types.V3,
+			Metadata: types.Metadata{
+				Name:    "fake-desktop",
+				Expires: &expiry,
+			},
+		},
+		Spec: types.WindowsDesktopSpecV3{
+			Addr:   "127.0.0.1:8090",
+			Domain: "test",
+			HostID: "fake-desktop",
+		},
+	})
+	require.NoError(t, err)
+
+	_, err = leaf.Process.GetAuthServer().UpsertDatabaseServer(ctx, &types.DatabaseServerV3{
+		Kind:    types.KindDatabaseServer,
+		Version: types.V3,
+		Metadata: types.Metadata{
+			Name:    "fake-db",
+			Expires: &expiry,
+		},
+		Spec: types.DatabaseServerSpecV3{
+			Hostname: "db",
+			HostID:   "fake-db",
+			Database: &types.DatabaseV3{
+				Kind:    types.KindDatabase,
+				Version: types.V3,
+				Metadata: types.Metadata{
+					Name: "dbdb",
+					Labels: map[string]string{
+						"env": "prod",
+					},
+				},
+				Spec: types.DatabaseSpecV3{
+					Protocol: "test-protocol",
+					URI:      "test-uri",
+				},
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	// Create the trusted cluster.
+	trustedClusterToken := "trusted-cluster-token"
+	tokenResource, err := types.NewProvisionToken(trustedClusterToken, []types.SystemRole{types.RoleTrustedCluster}, time.Time{})
+	require.NoError(t, err)
+	err = main.Process.GetAuthServer().UpsertToken(ctx, tokenResource)
+	require.NoError(t, err)
+	trustedCluster := main.AsTrustedCluster(trustedClusterToken, types.RoleMap{
+		{Remote: teleport.PresetAccessRoleName, Local: []string{teleport.PresetAccessRoleName, leafRole.GetName()}},
+	})
+
+	require.NoError(t, main.Start())
+	require.NoError(t, leaf.Start())
+	require.NoError(t, trustedCluster.CheckAndSetDefaults())
+
+	// Persist the trusted cluster resource.
+	helpers.TryCreateTrustedCluster(t, leaf.Process.GetAuthServer(), trustedCluster)
+	helpers.WaitForTunnelConnections(t, main.Process.GetAuthServer(), "leaf", 1)
+
+	// Wait for both clusters to see each other via reverse tunnels.
+	require.Eventually(t, helpers.WaitForClusters(main.Tunnel, 1), 10*time.Second, 1*time.Second,
+		"Two clusters do not see each other: tunnels are not working.")
+	require.Eventually(t, helpers.WaitForClusters(leaf.Tunnel, 1), 10*time.Second, 1*time.Second,
+		"Two clusters do not see each other: tunnels are not working.")
+
+	// Get the resources from the web api and validate the logins appear as expected.
+	clt, err := main.NewWebClient(helpers.ClientConfig{
+		Login:    "alice",
+		Password: "password",
+		Cluster:  helpers.Site,
+		Host:     Loopback,
+		Port:     helpers.Port(t, main.SSH),
+	})
+	require.NoError(t, err)
+
+	resources, err := clt.GetResources("leaf")
+	require.NoError(t, err)
+	require.Len(t, resources, 3)
+
+	for _, resource := range resources {
+		switch resource["kind"] {
+		case types.KindNode:
+			// Note: the ssh logins include "moe" from the root-role because it gets added
+			// as a valid principal to the user SSH certificate, and all valid principals get
+			// added to the user's login traits.
```

Comment:
```suggestion
			// Note: the ssh logins include "moe" from the root-role because:
			// - all valid logins get added to the SSH certificate ValidPrincipals
			// - the ValidPrincipals overwrites the internal logins trait in the leaf (for historical reasons)
			// - the user has the access role in the leaf, which expands the internal logins trait
```